dan_layer/transaction/src/transaction.rs

```diff
@@ -146,18 +153,36 @@ impl Eq for Transaction {}
 
 #[derive(Debug, Clone, Default, Deserialize, Serialize, Eq, PartialEq)]
 pub struct TransactionMeta {
+    required_inputs: Vec<SubstateRequirement>,
     involved_objects: HashMap<ShardId, SubstateChange>,
```

Comment:
Think we can remove involved_objects in future and just have explicit `<Substate, Option<Version>>` inputs. VNs can translate to shard ids in the mempool

Reply:
I agree, the `involved_objects` use ShardIds which is a bit inconvenient (makes it hard to interpret), we may need a new field with a new type for auto-filled inputs or outputs without the optional version (something like `<Substate, u32>`)

Reply:
:+1: Makes sense, the required_inputs are in the sig challenge and the autofilled_inputs are included by some notary (e.g. maybe indexer and/or maybe a VN), which signs the additional inputs/outputs and these are "validated" by VNs - that is, the VN can quickly validate that these should have been included without running the template.